sdk/core/core-amqp/src/errors.ts

```diff
@@ -553,6 +553,18 @@ export function translate(err: AmqpError | Error): MessagingError {
 
   let error: MessagingError = err as MessagingError;
 
+  // InsufficientCreditError occurs when the number of credits available on Rhea link is insufficient.
+  // Since reasons for such shortage can be transient such as for pending delivery of messages, this is treated as a retryable error.
+  if (
+    // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
+    // Must do a name check until InsufficientCreditError is updated, and that doesn't break compatibility
+    // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    (err as Error).name === "InsufficientCreditError"
+  ) {
+    error.retryable = true;
+    return error;
+  }
```

Comment:
This section of the `translate()` method is getting very verbose with the repetition of comments on why we do a `name` check instead of a `instanceof` check. Please refactor appropriately for brevity.

Reply:
Addressed this and below comments as well